packages/components/inputs/input-utils/src/multiline-input/multiline-input.tsx

```diff
@@ -1,39 +1,65 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import requiredIf from 'react-required-if';
-import TextareaAutosize from 'react-textarea-autosize';
+import React, { ChangeEventHandler } from 'react';
+import { useTheme } from '@emotion/react';
+import TextareaAutosize, {
+  TextareaHeightChangeMeta,
+} from 'react-textarea-autosize';
 import { filterDataAttributes } from '@commercetools-uikit/utils';
 import { getTextareaStyles } from './multiline-input.styles';
 
 const MIN_ROW_COUNT = 1;
 
-const MultilineInput = (props) => {
+export type TMultiLineInputProps = {
+  autoComplete?: string;
+  className?: string;
+  hasError?: boolean;
+  hasWarning?: boolean;
+  id?: string;
+  isAutofocussed?: boolean;
+  isDisabled?: boolean;
+  isReadOnly?: boolean;
+  name?: string;
+  onBlur?: ChangeEventHandler;
+  onChange?: ChangeEventHandler;
+  onFocus?: ChangeEventHandler;
+  placeholder?: string;
+  value: string;
+  isOpen: boolean;
+  onHeightChange?: (height: number, rowCount: number) => void;
+};
+
+const MultilineInput = (props: TMultiLineInputProps) => {
+  const theme = useTheme();
   const { onHeightChange } = props;
   const ref = React.useRef();
-  const handleHeightChange = React.useCallback(
-    (height, meta) => {
-      const rowCount = Math.floor(ref.current.scrollHeight / meta.rowHeight);
-      onHeightChange(height, rowCount);
+  const handleHeightChange = React.useCallback<
+    (height: number, meta?: TextareaHeightChangeMeta) => void
+  >(
+    (height: number, meta?: TextareaHeightChangeMeta) => {
+      // @ts-ignore
+      const rowCount = Math.floor(ref?.current?.scrollHeight / meta?.rowHeight);
+      if (onHeightChange) {
+        onHeightChange(height, rowCount);
+      }
     },
     [ref, onHeightChange]
   );
   return (
     <TextareaAutosize
+      // @ts-ignore
       ref={ref}
```

Comment:
The `TextareaAutosize` exports `React.forwardRef`. What error do you get?

https://github.com/Andarist/react-textarea-autosize/blob/58c9807645dceb9034b733d9cb63aa66df2364b0/src/index.tsx#L101

Reply:
it looks like a mismatch on the type of `useRef` and what is accepted as `ref` on the component

![image](https://user-images.githubusercontent.com/462507/116212306-fda51000-a744-11eb-8dbf-a0c80fb7a20b.png)

Reply:
Then use the correct type 😜

Reply:
which one then? 😄 

we are passing a `ref` into a component that extends `TextareaHTMLAttributes` with `IntrinsicAttributes`, none accepts `MutableRefObject`..
maybe I misunderstood something

Reply:
I just rubber ducked myself 🦆

Reply:
https://github.com/commercetools/ui-kit/pull/1867/commits/e9737e4067132362ecdca027375dfa1bd15f066f